In [ ]:
import pandas as pd

# Load the dataset
file_path = '/content/sentiment_data.csv'
sentiment_data = pd.read_csv(file_path)

# Check the available columns
print(sentiment_data.columns)

Index(['sentiment', 'Sentence'], dtype='object')


In [ ]:
# Clean the 'Sentence' column and create 'cleaned_text'
if 'Sentence' in sentiment_data.columns:
    sentiment_data['cleaned_text'] = sentiment_data['Sentence'].str.lower().str.replace('[^\w\s]', '', regex=True)
else:
    raise ValueError("Expected column 'Sentence' not found for text cleaning.")

# Display the first few rows to verify the cleaning
print(sentiment_data[['Sentence', 'cleaned_text']].head())

                                            Sentence  \
0  According to Gran   the company has no plans t...   
1  For the last quarter of 2010   Componenta 's n...   
2  In the third quarter of 2010   net sales incre...   
3  Operating profit rose to EUR 13.1 mn from EUR ...   
4  Operating profit totalled EUR 21.1 mn   up fro...   

                                        cleaned_text  
0  according to gran   the company has no plans t...  
1  for the last quarter of 2010   componenta s ne...  
2  in the third quarter of 2010   net sales incre...  
3  operating profit rose to eur 131 mn from eur 8...  
4  operating profit totalled eur 211 mn   up from...  


In [ ]:
sentiment_data.to_csv('cleaned_sentiment_data.csv', index=False)

In [ ]:
# Display the dataset structure
print(sentiment_data.head())

# Check for missing values
sentiment_data.dropna(inplace=True)

# Clean text column if necessary (e.g., 'cleaned_text' column)
if 'cleaned_text' in sentiment_data.columns:
    sentiment_data['cleaned_text'] = sentiment_data['cleaned_text'].str.lower().str.replace('[^\w\s]', '')
else:
    raise ValueError("Expected column 'cleaned_text' not found.")

  sentiment                                           Sentence  \
0         0  According to Gran   the company has no plans t...   
1         1  For the last quarter of 2010   Componenta 's n...   
2         1  In the third quarter of 2010   net sales incre...   
3         1  Operating profit rose to EUR 13.1 mn from EUR ...   
4         1  Operating profit totalled EUR 21.1 mn   up fro...   

                                        cleaned_text  
0  according to gran   the company has no plans t...  
1  for the last quarter of 2010   componenta s ne...  
2  in the third quarter of 2010   net sales incre...  
3  operating profit rose to eur 131 mn from eur 8...  
4  operating profit totalled eur 211 mn   up from...  


In [ ]:
!pip install transformers torch tensorflow keras

In [ ]:
from transformers import pipeline

# Load Hugging Face LLM
text_generator = pipeline("text-generation", model="gpt2")

def chatbot_response(prompt):
    # Generate response from Hugging Face
    response = text_generator(prompt, max_length=100, num_return_sequences=1)
    return response[0]['generated_text']


/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/665 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/548M [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/124 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/26.0 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/1.04M [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

Device set to use cpu


In [ ]:
import numpy as np
from sklearn.preprocessing import MinMaxScaler
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, LSTM

In [ ]:
# Generate synthetic timestamps with a higher frequency (hourly or minutely)
sentiment_data['timestamp'] = pd.date_range(start="2024-01-01", periods=len(sentiment_data), freq='T')  # Minutely frequency

# Sort the data by the synthetic timestamp
sentiment_data.sort_values(by='timestamp', inplace=True)

# Display the first few rows
print(sentiment_data[['timestamp', 'Sentence']].head())

            timestamp                                           Sentence
0 2024-01-01 00:00:00  According to Gran   the company has no plans t...
1 2024-01-01 00:01:00  For the last quarter of 2010   Componenta 's n...
2 2024-01-01 00:02:00  In the third quarter of 2010   net sales incre...
3 2024-01-01 00:03:00  Operating profit rose to EUR 13.1 mn from EUR ...
4 2024-01-01 00:04:00  Operating profit totalled EUR 21.1 mn   up fro...


<ipython-input-8-6b6f4798a018>:2: FutureWarning: 'T' is deprecated and will be removed in a future version, please use 'min' instead.
  sentiment_data['timestamp'] = pd.date_range(start="2024-01-01", periods=len(sentiment_data), freq='T')  # Minutely frequency


In [ ]:
# Generate random timestamps within a specific range
start_date = pd.Timestamp("2024-01-01")
end_date = pd.Timestamp("2024-12-31")
sentiment_data['timestamp'] = pd.to_datetime(
    np.random.uniform(start_date.value, end_date.value, len(sentiment_data))
).sort_values()

# Display the first few rows
print(sentiment_data[['timestamp', 'Sentence']].head())

                      timestamp  \
0 2024-01-01 00:05:38.175014912   
1 2024-01-01 00:12:05.286381568   
2 2024-01-01 00:20:14.733343232   
3 2024-01-01 00:26:12.543404544   
4 2024-01-01 00:30:13.202571264   

                                            Sentence  
0  According to Gran   the company has no plans t...  
1  For the last quarter of 2010   Componenta 's n...  
2  In the third quarter of 2010   net sales incre...  
3  Operating profit rose to EUR 13.1 mn from EUR ...  
4  Operating profit totalled EUR 21.1 mn   up fro...  


In [ ]:
# Ensure the dataset contains a timestamp and market data (e.g., 'score')
sentiment_data['timestamp'] = pd.to_datetime(sentiment_data['timestamp'])
sentiment_data.sort_values(by='timestamp', inplace=True)

In [ ]:
import pandas as pd
from sklearn.preprocessing import MinMaxScaler

# Check available columns
print(sentiment_data.columns)

# Create a numerical score based on sentiment values (if needed)
# Assuming sentiment has values like "positive", "negative", "neutral"
sentiment_mapping = {'positive': 1, 'neutral': 0.5, 'negative': 0}
sentiment_data['score'] = sentiment_data['sentiment'].map(sentiment_mapping)

# Handle missing or unmapped sentiment values
if sentiment_data['score'].isnull().any():
    print("Warning: Some sentiment values are not mapped. Assigning default score 0.")
    sentiment_data['score'].fillna(0, inplace=True)

# Add synthetic timestamps (if not already done)
if 'timestamp' not in sentiment_data.columns:
    sentiment_data['timestamp'] = pd.date_range(
        start="2024-01-01", periods=len(sentiment_data), freq='T'
    )

# Extract features and scale
market_data = sentiment_data[['timestamp', 'score']]
scaler = MinMaxScaler(feature_range=(0, 1))
market_data['scaled_score'] = scaler.fit_transform(market_data[['score']])

# Display the scaled data
print(market_data.head())


Index(['sentiment', 'Sentence', 'cleaned_text', 'timestamp'], dtype='object')
                      timestamp  score  scaled_score
0 2024-01-01 00:05:38.175014912    0.0           0.0
1 2024-01-01 00:12:05.286381568    0.0           0.0
2 2024-01-01 00:20:14.733343232    0.0           0.0
3 2024-01-01 00:26:12.543404544    0.0           0.0
4 2024-01-01 00:30:13.202571264    0.0           0.0


<ipython-input-11-ba8347c06af4>:15: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  sentiment_data['score'].fillna(0, inplace=True)
<ipython-input-11-ba8347c06af4>:26: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  market_data['scaled_score'] = scaler.fit_transform(market_data[['score']])


In [ ]:
# Extract features and scale
market_data = sentiment_data[['timestamp', 'score']]
scaler = MinMaxScaler(feature_range=(0, 1))
scaled_data = scaler.fit_transform(market_data[['score']])

In [ ]:
# Create time series dataset
def create_time_series(data, time_step=10):
    X, y = [], []
    for i in range(len(data) - time_step - 1):
        X.append(data[i:(i + time_step), 0])
        y.append(data[i + time_step, 0])
    return np.array(X), np.array(y)

In [ ]:
time_step = 10
X, y = create_time_series(scaled_data, time_step)

In [ ]:
# Reshape to LSTM input format
X = X.reshape(X.shape[0], X.shape[1], 1)

# Split into training and testing data
train_size = int(len(X) * 0.8)
X_train, X_test = X[:train_size], X[train_size:]
y_train, y_test = y[:train_size], y[train_size:]

In [ ]:
# Define LSTM model
model = Sequential([
    LSTM(50, return_sequences=True, input_shape=(time_step, 1)),
    LSTM(50, return_sequences=False),
    Dense(25),
    Dense(1),
])

/usr/local/lib/python3.10/dist-packages/keras/src/layers/rnn/rnn.py:204: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(**kwargs)


In [ ]:
# Compile model
model.compile(optimizer='adam', loss='mean_squared_error')

In [ ]:
# Train model
model.fit(X_train, y_train, batch_size=32, epochs=10, validation_data=(X_test, y_test))

Epoch 1/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 80s 25ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 2/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 53s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 3/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 41s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 4/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 39s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 5/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 42s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 6/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 40s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 7/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 42s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 8/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 40s 15ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 9/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 40s 15ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 10/10
2720/2720 ━━━━━━━━━━━━━━━━━━━━ 40s 14ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00


In [ ]:
# Forecast future trends
def forecast_trend(input_data, n_steps=10):
    predictions = []
    current_input = input_data[-time_step:].reshape(1, time_step, 1)
    for _ in range(n_steps):
        next_value = model.predict(current_input, verbose=0)[0, 0]
        predictions.append(next_value)
        current_input = np.append(current_input[:, 1:, :], [[next_value]], axis=1)
    return scaler.inverse_transform(np.array(predictions).reshape(-1, 1))

In [ ]:
from transformers import GPT2LMHeadModel, GPT2Tokenizer

# Load GPT-2 model and tokenizer
model_name = "gpt2"
tokenizer = GPT2Tokenizer.from_pretrained(model_name)
model = GPT2LMHeadModel.from_pretrained(model_name)

In [ ]:
def market_chatbot(user_input):

    inputs = tokenizer(user_input, return_tensors="pt")

    output = model.generate(
        inputs["input_ids"],
        max_length=100,
        num_return_sequences=1,
        temperature=0.8,         # Ajoute de la créativité
        top_p=0.9,              # Limite les choix à des tokens probables
        repetition_penalty=1.2  # Réduit les répétitions
    )

    # Decode the generated text and return
    response = tokenizer.decode(output[0], skip_special_tokens=True)
    return response

In [ ]:
# Example Usage

print(market_chatbot("How can someone start investing with a small budget?"))

/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py:628: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0.8` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`.
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py:633: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`.
  warnings.warn(
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpec

How can someone start investing with a small budget?
The answer is simple: you need to be able and willing to invest in the right things. You don't have time for spending money on fancy gadgets or flashy products, but rather your own personal style of living that will make it easier than ever before (and more fun). The best way to do this would probably not involve buying an expensive car – which I'm sure many people are already doing anyway! But if there's one thing we all


In [ ]:
from transformers import GPT2LMHeadModel, GPT2Tokenizer

# Chemin pour sauvegarder le modèle
save_path = "./finetuned_gpt2"

# Sauvegarder le modèle et le tokenizer
model.save_pretrained(save_path)
tokenizer.save_pretrained(save_path)

print(f"Model and tokenizer saved to {save_path}")


Model and tokenizer saved to ./finetuned_gpt2


In [ ]:
import pandas as pd
import sqlite3
import kagglehub

# Télécharger le dataset Kaggle
path = kagglehub.dataset_download("mysarahmadbhat/inc-5000-companies")

# Charger les données dans un DataFrame
csv_file = "/content/INC 5000 Companies 2019.csv"  # Remplacez par le nom exact du fichier téléchargé
data = pd.read_csv(csv_file)

# Charger les données dans une base de données SQLite
conn = sqlite3.connect("stock_db.sqlite")
data.to_sql("companies", conn, if_exists="replace", index=False)

print("Database successfully created with the table 'companies'")


Database successfully created with the table 'companies'


In [ ]:
from transformers import GPT2LMHeadModel, GPT2Tokenizer
# Chemin où le modèle est sauvegardé
load_path = "finetuned_gpt2"

# Charger le modèle et le tokenizer
tokenizer = GPT2Tokenizer.from_pretrained(load_path)
model = GPT2LMHeadModel.from_pretrained(load_path)

print("Finetuned model and tokenizer loaded successfully!")


In [ ]:
!pip install openai==0.28

In [ ]:
from openai import AzureOpenAI

In [ ]:
from openai import AzureOpenAI

def generate_sql_query(user_prompt):
    # Configuration
    endpoint = "https://moham-m5wb6cap-eastus2.cognitiveservices.azure.com/openai/deployments/gpt-35-turbo/chat/completions?api-version=2024-08-01-preview"
    deployment_name = "gpt-35-turbo"
    api_key = "BRklaIIEnrE5oYzxJP8strvVD94DH38tRVjXHJkiNYiRIHhwIhzPJQQJ99BAACHYHv6XJ3w3AAAAACOGpenY"

    # Initialize Azure OpenAI client using API key
    client = AzureOpenAI(
        azure_endpoint=endpoint,
        api_key=api_key,
        api_version="2024-08-01-preview",
    )

    # Table columns list with "revenue (Millions)" as the correct name
    table_columns = "[rank, profile, name, url, state, revenue (Millions), growth_%, industry, workers, previous_workers, founded, yrs_on_list, metro, city]"

    # Updated system and user prompt with correct column name 'revenue (Millions)'
    chat_prompt = [
        {
            "role": "system",
            "content": f"The table 'companies' contains the following columns: {table_columns}. Use the correct column names in your SQL query."
        },
        {
            "role": "user",
            "content": f"Convert this user query to a SQL query: {user_prompt}. Ensure you use 'revenue (Millions)' as the column name for revenue."
        }
    ]

    # Generate chat completion
    completion = client.chat.completions.create(
        model=deployment_name,
        messages=chat_prompt,
        max_tokens=800,
        temperature=0.7,
        top_p=0.95,
        frequency_penalty=0,
        presence_penalty=0,
        stop=None,
        stream=False
    )

    # Extract the query text from the API response and return it
    return completion.choices[0].message.content.strip()




  # SELECT name, "revenue (Millions)", industry
  # FROM companies
  # WHERE "revenue (Millions)" > 20;

In [ ]:
# code to reorganise the the db

import pandas as pd
import sqlite3
import kagglehub  # Assuming you have a function to download datasets

# Download the dataset from Kaggle
path = kagglehub.dataset_download("mysarahmadbhat/inc-5000-companies")

# Load the data into a DataFrame
csv_file = "/content/INC 5000 Companies 2019.csv"  # Replace with the correct filename
data = pd.read_csv(csv_file)

# Check if 'revenue' exists and preprocess it
if 'revenue' in data.columns:  # Use lowercase 'revenue'
    def convert_revenue(value):
        """Convert revenue strings to numeric values in millions."""
        value = value.strip()  # Remove leading/trailing whitespace
        if 'Million' in value:
            return float(value.replace(' Million', '').replace(',', ''))
        elif 'Billion' in value:
            return float(value.replace(' Billion', '').replace(',', '')) * 1000
        else:
            return None  # Handle unexpected formats

    # Apply the conversion
    data['revenue (Millions)'] = data['revenue'].apply(convert_revenue)
    # Drop the original 'revenue' column
    data = data.drop(columns=['revenue'])

# Display the DataFrame after preprocessing
print("Transformed DataFrame:")
print(data.head())

# Save the transformed data into a new SQLite database
conn = sqlite3.connect("stock_db.sqlite")
data.to_sql("companies", conn, if_exists="replace", index=False)

print("Database successfully created with the table 'companies' containing 'revenue (Millions)' as numeric values.")


Transformed DataFrame:
   rank                                   profile               name  \
0     1      https://www.inc.com/profile/freestar           Freestar   
1     2   https://www.inc.com/profile/freightwise        FreightWise   
2     3  https://www.inc.com/profile/ceces-veggie  Cece's Veggie Co.   
3     4      https://www.inc.com/profile/ladyboss           LadyBoss   
4     5        https://www.inc.com/profile/perpay             Perpay   

                         url state    growth_%                      industry  \
0        http://freestar.com    AZ  36680.3882       Advertising & Marketing   
1  http://freightwisellc.com    TN  30547.9317    Logistics & Transportation   
2   http://cecesveggieco.com    TX  23880.4852               Food & Beverage   
3        http://ladyboss.com    NM  21849.8925  Consumer Products & Services   
4          http://perpay.com    PA  18166.4070                        Retail   

   workers  previous_workers  founded  yrs_on_list         metr

In [ ]:
'```sql SELECT name, "revenue (Millions)", industry FROM companies WHERE "revenue (Millions)" BETWEEN 800 AND 900; ```'.replace('`', '').replace('\n', ' ').replace('sql', ' ').strip()

'SELECT name, "revenue (Millions)", industry FROM companies WHERE "revenue (Millions)" BETWEEN 800 AND 900;'

In [ ]:
def market_chatbot(user_prompt):
    # Generate the SQL query based on user input
    sql_query = generate_sql_query(user_prompt).replace('`', '').replace('\n', ' ').replace('sql', ' ').strip()

    # Connect to the SQLite database
    conn = sqlite3.connect("stock_db.sqlite")
    cursor = conn.cursor()

    # Execute the generated SQL query
    cursor.execute(sql_query)

    # Fetch the results (retrieved data)
    retrieved_data = cursor.fetchall()

    # Process the retrieved data (convert it into a string format for the prompt)
    retrieved_data_str = ""
    for row in retrieved_data:
        retrieved_data_str += f"- {row[0]}: {row[1]} Million (Industry: {row[2]})\n"

    # Close the connection
    conn.close()

    # Create a focused prompt to guide the model
    prompt = (
        f"Here are the companies with annual revenues between 800 Million and 900 Million:\n"
        f"{retrieved_data_str}\n"
        "Please provide a concise, factual, and direct response without any additional commentary or information."
    )

    # Ensure proper tokenizer setup
    if tokenizer.pad_token is None:
        tokenizer.pad_token = tokenizer.eos_token  # Avoid padding-related errors

    # Tokenize the input with padding and truncation
    inputs = tokenizer(prompt, return_tensors="pt", padding=True, truncation=True)

    # Generate output based on the prompt and data
    output = model.generate(
        inputs["input_ids"],
        attention_mask=inputs["attention_mask"],
        pad_token_id=tokenizer.eos_token_id,
        max_new_tokens=150,
        num_return_sequences=1,
        do_sample=False,  # Use deterministic output generation
        temperature=0.7,  # Lower temperature for focused responses
        top_p=0.9,  # Keep top-p sampling to ensure coherence
        repetition_penalty=1.2
    )

    # Decode and return the generated response
    response = tokenizer.decode(output[0], skip_special_tokens=True)

    # Remove any extra unwanted text
    response = response.replace("Please provide a concise, factual, and direct response without any additional commentary or information.", "").strip()

    return response




* GPT-3.5 generates the SQL query. ✅

* We need to execute the query to retrieve data from the database (stock_db.sqlite) ✅

* GPT-2 can process the data to generate valid replies. ✅






In [ ]:
market_chatbot("List the names, revenues, and industries of companies that have an annual revenue between 800 Million and 900 Million.")

'Here are the companies with annual revenues between 800 Million and 900 Million:\n- Nolan Transportation Group: 811.9 Million (Industry: Logistics & Transportation)\n- Young Automotive Group: 886.5 Million (Industry: Retail)\n- Pro Mach: 872.1 Million (Industry: Manufacturing)\n- Guaranteed Rate: 802.6 Million (Industry: Financial Services)'

Here are the companies with annual revenues between 800 Million and 900 Million:
- Nolan Transportation Group: 811.9 Million (Industry: Logistics & Transportation)
- Young Automotive Group: 886.5 Million (Industry: Retail)
- Pro Mach: 872.1 Million (Industry: Manufacturing)
- Guaranteed Rate: 802.6 Million (Industry: Financial Services)

In [ ]:
!pip install flask-ngrok


In [ ]:
!pip install pyngrok

In [ ]:
!ngrok authtoken 2pCKj4hrIGwDWVBkVk5pgFG5xXs_5oTWBV4ywLHFCkrHtad56

Authtoken saved to configuration file: /root/.config/ngrok/ngrok.yml


In [ ]:
from pyngrok import ngrok

# Open a tunnel to the Flask app on port 5000
public_url = ngrok.connect(5000)
print('Flask app is available at:', public_url)


Flask app is available at: NgrokTunnel: "https://93c4-34-23-121-109.ngrok-free.app" -> "http://localhost:5000"


In [ ]:
from flask import Flask, request, jsonify
from transformers import GPT2Tokenizer, GPT2LMHeadModel

app = Flask(__name__)

# Load the GPT-2 model and tokenizer
model_name = "gpt2"
tokenizer = GPT2Tokenizer.from_pretrained(model_name)
model = GPT2LMHeadModel.from_pretrained(model_name)

@app.route('/chat', methods=['POST'])
def chat():
    prompt = request.json.get('prompt')
    if not prompt:
        return jsonify({'error': 'No prompt provided'}), 400

    # Call the market_chatbot function to get the response
    response = market_chatbot(prompt)

    return jsonify({'response': response})

if __name__ == '__main__':
    app.run(port=5000)


 * Serving Flask app '__main__'
 * Debug mode: off


INFO:werkzeug:WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
INFO:werkzeug:Press CTRL+C to quit
INFO:werkzeug:127.0.0.1 - - [15/Jan/2025 01:35:43] "POST /chat HTTP/1.1" 415 -
INFO:werkzeug:127.0.0.1 - - [15/Jan/2025 01:40:26] "POST /chat HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [15/Jan/2025 01:58:51] "GET /chat' HTTP/1.1" 404 -
INFO:werkzeug:127.0.0.1 - - [15/Jan/2025 01:59:00] "GET /chat' HTTP/1.1" 404 -
